### PR TITLE
add test on Alpine and Nanoserver using busybox

### PR DESCRIPTION
**Pull request by hemnstill**
- add alpine tests using bsdtar
- no need to checkout sources and install bash when run makeself.run
- git submodules as single step:
```
 with:
   submodules: 'true'
```

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -20,50 +20,55 @@ jobs:
...
-      uses: actions/download-artifact@v2
+    - uses: actions/download-artifact@v2
       with:
         name: Makeself-Build-Windows
         path: windows
     - name: Test on FreeBSD
       uses: vmactions/freebsd-vm@v0.1.5
       with:
         usesh: true
-        prepare: pkg install -y pbzip2 bash
```
Comment (megastep) [inline .github/workflows/build.yml:66]: Do you really need to skip install of these packages, especially pbzip2?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -75,36 +80,87 @@ jobs:
...
-      uses: actions/download-artifact@v2
+    - uses: actions/download-artifact@v2
       with:
         name: Makeself-Build-Windows
         path: windows
     - name: Test on Solaris
       uses: vmactions/solaris-vm@v0.0.3
       with:
         mem: 2048
-        prepare: pkgutil -y -i pbzip2
```
Comment (megastep) [inline .github/workflows/build.yml:101]: Same here

**Review by megastep — COMMENTED** [review]
This looks pretty good; just a few concerns as to why you removed the install of pbzip2 as it can be tested for in the test suite, even if those are not standard packages.

File: .github/workflows/build.yml (hunk 1 above)
Comment (hemnstill) [reply .github/workflows/build.yml:66]: revert pbzip2 

I thought that this way we can check the unpacking on a clean system. But this changes is not required for this task.

File: .github/workflows/build.yml (hunk 2 above)
Comment (hemnstill) [reply .github/workflows/build.yml]: revert pbzip2

**Comment by megastep** [comment]
Thanks!

**Pull request merged**